src/components/TasksTable/Columns.tsx

```diff
@@ -0,0 +1,155 @@
+import { LGColumnDef } from "@leafygreen-ui/table";
+import Tooltip from "@leafygreen-ui/tooltip";
+import pluralize from "pluralize";
+import { ConditionalWrapper } from "components/ConditionalWrapper";
+import { StyledRouterLink } from "components/styles";
+import TaskStatusBadge from "components/TaskStatusBadge";
+import { TreeDataEntry } from "components/TreeSelect";
+import { getVariantHistoryRoute } from "constants/routes";
+import { mergeTaskVariant } from "constants/task";
+import { zIndex } from "constants/tokens";
+import { TaskSortCategory } from "gql/generated/types";
+import { TaskStatus } from "types/task";
+import { TaskLink } from "./TaskLink";
+import { TaskTableInfo } from "./types";
+
+export const getColumnsTemplate = ({
+  baseStatusOptions = [],
+  isPatch = false,
+  onClickTaskLink = () => {},
+  showTaskExecutionLabel = false,
+  statusOptions = [],
+}: {
+  baseStatusOptions?: TreeDataEntry[];
+  isPatch?: boolean;
+  onClickTaskLink?: (taskId: string) => void;
+  showTaskExecutionLabel?: boolean;
+  statusOptions?: TreeDataEntry[];
+}): LGColumnDef<TaskTableInfo>[] => [
+  {
+    header: "Name",
+    accessorKey: "displayName",
+    id: TaskSortCategory.Name,
+    cell: ({
+      getValue,
+      row: {
+        original: { execution, id },
+      },
+    }): JSX.Element => (
+      <TaskLink
+        execution={execution}
+        onClick={onClickTaskLink}
+        showTaskExecutionLabel={showTaskExecutionLabel}
+        taskId={id}
+        taskName={getValue() as string}
+      />
+    ),
+    meta: {
+      search: {
+        "data-cy": "task-name-filter-popover",
+        placeholder: "Task name regex",
+      },
+    },
```

Comment:
Should this be removed since task name isn't a searchable column for the execution tasks table? Or will it be used later?

Reply:
this column definition will be reused for the upcoming tables, and they'll be shown in those tables! (initially i wrote these changes in one big PR which got split up now into 3 PRs, so that's why they are here 😅)